R/beers-r6.R

```diff
@@ -0,0 +1,153 @@
+Beers_Interpolator <- R6::R6Class("Beers_Interpolator",
+
+  public = list(
+    b1 = NULL,
+    b2 = NULL,
+    bm = NULL,
+    bp = NULL,
+    bf = NULL,
+
+    initialize = function(b1, b2, bm, bp, bf) {
+      self$b1 <- b1
+      self$b2 <- b2
+      self$bm <- bm
+      self$bp <- bp
+      self$bf <- bf
+    },
+
+    interpolate = function(points) {
```

Comment:
I might be missing something here, but how is the domain of the required values specified?  Usually when I think of interpolation I have a set of values `x` and `y` and I want to interpolate onto some new set of values `x2` that lie within the range of `x`.  Here I don't see either a specification of `y` or a way of choosing which `x` values should be returned

Reply:
The interpolation is done between the points in the original list - it expands the list to 5-times as long, and gives in the gap. Beers is a very "one-purpose" algorithm...